File: class1_ROS_Basic/pubsub_example_pkg/scripts/simple_topic_publisher.py
#! /usr/bin/env python

# Import the Python library for ROS
import rospy
# Import the Int32 message from the std_msgs package
from std_msgs.msg import Int32             
from custom_messages.msg import Custom_topic

# Initiate a Node named 'topic_publisher'
rospy.init_node('topic_publisher')

# Create a Publisher object, that will publish on the /counter topic
# messages of type Int32
pub_1 = rospy.Publisher('/counter', Int32, queue_size=1)    
pub_2 = rospy.Publisher('/custom_topic', Custom_topic, queue_size=1)
# Set a publish rate of 2 Hz
rate = rospy.Rate(2)
# Create a variable of type Int32
count = Int32()
# Initialize 'count' variable
count.data = 0

custom_msg = Custom_topic()                        

# Create a loop that will go until someone stops the program execution
while not rospy.is_shutdown():
  # Publish the message within the 'count' variable
  print("count : ",count)
  print("custom_msg_a : ",custom_msg.a)
  print("custom_msg_b : ",custom_msg.b)
  # Increment 'count' variable
  count.data += 1
  custom_msg.a = count.data
  custom_msg.b = count.data
  pub_1.publish(count)
  pub_2.publish(custom_msg)
  # Make sure the publish rate maintains at 2 Hz
  rate.sleep()                             

File: class1_ROS_Basic/pubsub_example_pkg_c/src/simple_topic_publisher.cpp
#include <ros/ros.h>
#include <std_msgs/Int32.h>
#include <custom_messages/Custom_topic.h>

int main(int argc, char** argv) {

    ros::init(argc, argv, "topic_publisher");
    ros::NodeHandle nh;
    
    ros::Publisher pub = nh.advertise<std_msgs::Int32>("counter", 1000);
    ros::Publisher pub2 = nh.advertise<custom_messages::Custom_topic>("custom",1000);
    ros::Rate loop_rate(2);
    
    std_msgs::Int32 count;
    custom_messages::Custom_topic custom_msg;

    count.data = 0;
    custom_msg.a = 1;
    custom_msg.b = 2;
    
    while (ros::ok())
    {
        pub.publish(count);
        pub2.publish(custom_msg);
        ros::spinOnce();
        loop_rate.sleep();
        ++count.data;
    }
    
    return 0;
}